redis_hashring/__init__.py

```diff
@@ -293,30 +295,34 @@ def poll(self):
         last_cleanup = time.time()
         self.cleanup()
 
-        while True:
-            # Since Redis' listen method blocks, we use select to inspect the
-            # underlying socket to see if there is activity.
-            fileno = pubsub.connection._sock.fileno()
-            timeout = max(0, POLL_INTERVAL - (time.time() - last_heartbeat))
-            r, w, x = self._select([fileno], [], [], timeout)
-            if fileno in r:
-                next(gen)
-                self.update()
-
-            last_heartbeat = time.time()
-            self.heartbeat()
-
-            now = time.time()
-            if now - last_cleanup > CLEANUP_INTERVAL:
-                last_cleanup = now
-                self.cleanup()
+        try:
+            while self._run_polling:
```

Comment:
Since `POLL_INTERVAL` is 10 seconds, does it mean that we may have to wait up to 10 seconds for it to shut down cleanly? Is there any way we can set a flag so that `select()` returns immediately when we do a shutdown?